BPOR/BPOR.Rms/Controllers/FilterController.cs

```diff
@@ -87,9 +100,60 @@ private VolunteerFilterViewModel ClearFilters(VolunteerFilterViewModel model)
         return new VolunteerFilterViewModel { StudyId = model.StudyId };
     }
 
+    private async Task PopulateFilterIndexDataAsync(VolunteerFilterViewModel model, CancellationToken cancellationToken)
+    {
+        if (model.StudyId.HasValue)
+        {
+            var study = await context.Studies
+                .Where(s => s.Id == model.StudyId)
+                .Select(s => new { s.StudyName, s.CpmsId, s.IsRecruitingIdentifiableParticipants })
+                .FirstOrDefaultAsync(cancellationToken);
+
+            if (study != null)
+            {
+                model.StudyName = study.StudyName;
+                model.StudyCpmsId = study.CpmsId;
+                model.ShowRecruitedFilter = study.IsRecruitingIdentifiableParticipants;
+                model.ShowPreferredContactFilter = study.IsRecruitingIdentifiableParticipants;
+            }
+        }
+
+        if (model.VolunteersPreferredContactItems == null || !model.VolunteersPreferredContactItems.Any())
+        {
+            model.VolunteersPreferredContactItems = VolunteerFilterViewModel.SetVolunteersPreferredContactItems();
+        }
+
+        if (model.VolunteersContactedItems == null || !model.VolunteersContactedItems.Any())
+        {
+            model.VolunteersContactedItems = VolunteerFilterViewModel.SetVolunteersContactedItems();
+        }
+
+        if (model.VolunteersRecruitedItems == null || !model.VolunteersRecruitedItems.Any())
+        {
+            model.VolunteersRecruitedItems = VolunteerFilterViewModel.SetVolunteersRecruitedItems();
+        }
+
+        if (model.VolunteersCompletedRegistrationItems == null || !model.VolunteersCompletedRegistrationItems.Any())
+        {
+            model.VolunteersCompletedRegistrationItems = VolunteerFilterViewModel.SetVolunteersCompletedRegistrationItems();
+        }
+
+        if (model.VolunteersRegisteredInterestItems == null || !model.VolunteersRegisteredInterestItems.Any())
+        {
+            model.VolunteersRegisteredInterestItems = VolunteerFilterViewModel.SetVolunteersRegisteredInterestItems();
+        }
+    }
+
     [HttpPost]
-    public async Task<IActionResult> SetupEmailCampaign(VolunteerFilterViewModel model, CancellationToken cancellationToken = default)
+    public async Task<IActionResult> SetupCampaign(VolunteerFilterViewModel model, CancellationToken cancellationToken = default)
     {
+        if (!model.SelectedVolunteersPreferredContact.Equals((int)ContactMethods.Email) && !model.SelectedVolunteersPreferredContact.Equals((int)ContactMethods.Letter) && model.ShowPreferredContactFilter)
+        {
+            TempData["ContactMethodError"] = "You must select a preferred contact method.";
```

Comment:
wrong validation message, see AC 2

https://nihr.atlassian.net/browse/CRNCC-2393
`Select if the volunteers preferred contact method is email or letter`